codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/DefaultAuthSchemeParamsSpec.java

```diff
@@ -77,16 +79,28 @@ private MethodSpec constructor() {
 
         if (authSchemeSpecUtils.generateEndpointBasedParams()) {
             parameters().forEach((name, model) -> {
-                if (authSchemeSpecUtils.includeParam(model, name)) {
+                if (authSchemeSpecUtils.includeParam(name)) {
                     String fieldName = authSchemeSpecUtils.paramMethodName(name);
-                    b.addStatement("this." + fieldName + " = builder." + fieldName);
+                    boolean isRequired = isParamRequired(model);
+                    if (isRequired) {
+                        b.addStatement(String.join(fieldName, "this.", " = $T.paramNotNull(builder.", ", \"", "\")"),
```

Comment:
nit: hard to read, but guess it's ok.

Reply:
Yeah, you're right, I will change it to:

```
                        b.addStatement("this.$1N = $2T.paramNotNull(builder.$1N, $1S)",
                                       fieldName, Validate.class);
```